v-next/hardhat-zod-utils/src/index.ts

```diff
@@ -164,7 +166,9 @@ export const resolvedConfigurationVariableSchema = z.object({
 /**
  * A Zod type to validate Hardhat's SensitiveString values.
  */
-export const sensitiveStringSchema = unionType(
+export const sensitiveStringSchema: z.ZodType<
+  string | z.infer<typeof configurationVariableSchema>
+> = unionType(
```

Comment:
This might seem unrelated, but it actually helped me improve the typings in `refineEdrNetworkUserConfig`, as some fields were being inferred as `any`.

A similar improvement can be made for `unionType` and `conditionalUnionType`, but those cases are a bit trickier, so I added TODOs to handle them in a separate PR.

Do you think we should create issues for those as well?